templates/raspbian_cattlepi/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers

```diff
@@ -168,6 +168,8 @@ cattlepi_check_apply_disk_layout()
             umount /boot
             # showtime: apply the requested layout
             sfdisk /dev/mmcblk0 < /tmp/sdlayout
```

Comment:
It may be good to use something like `udevadm settle` after this operation so ensure everything is as you expect.  See this answer on SO:  https://superuser.com/a/756005

Reply:
For completeness - they suggest something like:
`udevadm settle --timeout=15 --exit-if-exists=$PARTITION`

Reply:
is udevadm part of busybox?

Reply:
Oh yeah, good point! :cry: .... I keep forgetting we're not working inside a "full" linux system at this point.

Your implementation should work just fine.

Reply:
technically it's busybox + whatever we pull in. so we can pull in the any executable + libraries linked with it. the downside it that the size of the initramfs increases. so it's always a tradeoff

that being said, i've just tested this and it seems to work. a little bit more testing and I should be able to start on the CI harness :+1: exciting stuff